### DISCUSSION TITLE
Potential issue with StashKey on main branch

### DISCUSSION THREAD

**Discussion by BeyondEvil** [pytest-dev/pytest#11045] [Q&A]
I run a pretty comprehensive test suite for [pytest-selenium](https://github.com/pytest-dev/pytest-selenium/actions/runs/5101360157)

As part of that suite I pull in the main branch for `pytest`, so that I can catch potential deprecations etc.

Well, those tests are now [failing](https://github.com/pytest-dev/pytest-selenium/actions/runs/5101360157/jobs/9170136311#step:11:2407)

Lots of failures that I'm working my way through, but something that stood out was:

[KeyError: <_pytest.stash.StashKey object at 0x7fb5b038c820>](https://github.com/pytest-dev/pytest-selenium/actions/runs/5101360157/jobs/9170136311#step:11:141)

I don't have time right now to create a minimal repo case, hence the discussion and not a proper issue.

All tests pass for python 3.7 through 3.11 for `7.3.1`.

**Comment by bluetech** [comment]
Hi @BeyondEvil, I just tried running pytest-selenium tests locally using pytest main. I do get failures, but they also happen on 7.3.1 (I am probably missing some external dependencies). In particular I don't see the StashKey error. So I think we'd need some help with reproducing.

**Reply by BeyondEvil** [reply]
I switched to using docker containers, to minimize any environment issues/difference, but I'm still seeing a heap of failures on "main" compared to 7.3.1.

https://github.com/pytest-dev/pytest-selenium/actions/runs/5115205804/jobs/9196240776#step:8:140

I'll try to come up with a minimal repro case.

**Reply by bluetech** [reply]
My procedure for tracking down regressions is:

- `pip install -e path/to/pytest/clone`
- reproduce the problem
- start bisecting in the pytest repo until the regressing commit is found

It doesn't really require a minimal reproduction, just *some* reproduction that works locally

**Reply by BeyondEvil** [reply]
I think this might be a red herring.

In the "devel" tox testenv: https://github.com/pytest-dev/pytest-selenium/blob/master/tox.ini#LL37C1-L37C15

I use `pre_pip`, this means that pre-releases are being installed. Which in turn means that `pytest-html` which has seen a major refactoring is getting installed (4.0.0rc4). At this point, I _think_ there's where the problem lies.

Even tho, funilly enough, I can't repro it either (at least not using python 3.9) on my machine. With or without tox... 🤷‍♂️ 

I'll do some more experimenting.

**Comment by BeyondEvil** [comment]
The issue seems to arise from the `12-dev`-version of python that the python github actions provides. I can't reproduce it using any stable versions.

Seeing how that can be a rabbit hole of infinite depth, I'm not going to spend any time further troubleshooting. I don't see any issues with pytest or the plugin.
